### DISCUSSION TITLE
Idempotent redeployment

### DISCUSSION THREAD

**Discussion by renepajta** [Azure/data-landing-zone#158] [General] [answered]
Hi all,

is it possible to make "Deploy To Azure" button run the same deployment in idempotent way? 

Every component consists of multiple services and if something goes wrong, the all resources needs to be deleted and redeployed again. This makes it painful experience, especially in cases of retriable errors (e.g. Private Link registration). 

What do you think?

**Comment by marvinbuss** [comment, marked as answer]
Hi @renepajta,

With the name checks that we just introduced, this will not be possible. One option would be to disable them again. If you run into deployment issues with e.g. private endpoints, you can use the `Redeploy` button. We have seen some intermittent issues with Private Endpoints, which can be solved by deleting the failed endpoint and then clicking the redeploy button on the respective sub deployment:
![image](https://user-images.githubusercontent.com/34542414/128144045-0d30ac11-53dd-4e37-9606-5384b3073979.png)

Would this be an option for now?
In the meantime, we have to work with some of the PGs to make sure that these intermittent issues are solved.

**Reply by marvinbuss** [reply]
@renepajta We are also optimizing the sub-deployments now, so that you can trigger a redeployment through the process mentioned above. We are investigating these issues with the product teams, to make the deployments more reliable and smooth for our users.

**Reply by marvinbuss** [reply]
@renepajta I also made the respective parameters in the bicep modules optional in order to simplify the redeployment, if anything fails while deploying ESA (see https://github.com/Azure/data-landing-zone/pull/166). In the meantime, we also started an investigation to identify the root cause for some of these intermittent private endpoint issues.
The CPU and GPU cluster issue shown above should also be fixed by now. There was an issue in the Machine Learning Resource Provider causing these issues.
